tests/integration/tests/charm_tests/openstack_exporter.py

```diff
@@ -6,11 +6,141 @@
 import logging
 import unittest
 
+import yaml
+from zaza import model
+
 logger = logging.getLogger(__name__)
 
+APP_NAME = "openstack-exporter"
+SNAP_NAME = "golang-openstack-exporter"
+STATUS_TIMEOUT = 60
+
+
+class OpenstackExporterBaseTest(unittest.TestCase):
+    """Base Class for charm integration tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run setup for the test class."""
+        cls.leader_unit = model.get_lead_unit(APP_NAME)
```

Comment:
**nitpick:** 

There is a `run_on_leader` and `run_action_on_leader` function in zaza.
You probably want to replace how you run action and execute command on the leader unit.

Reply:
but there will be extra calls to get application -> list of units -> filter by leader for each call of `run_*_on_leader` though; while `get_leader_unit` is only called for each test class

Reply:
The cost of `get_leader_unit` should be low? I don't think we need to worry about that in functional test.

Reply:
I can do that, but I think it will be a waste of calling `run_on_leader` because we will need to get the leader unit anyways

Reply:
Done

Reply:
Could the leader change during the test run?